src/platforms/hosted/cmsis_dap.c

```diff
@@ -477,7 +477,7 @@ static void dap_mem_read(adiv5_access_port_s *ap, void *dest, uint32_t src, size
 	if (len == 0)
 		return;
 	align_e align = MIN(ALIGNOF(src), ALIGNOF(len));
-	DEBUG_WIRE("dap_mem_read @ %" PRIx32 " len %zu, align %d\n", src, len, align);
+	DEBUG_WIRE("dap_mem_read @ %" PRIx32 " len %zu, align %u\n", src, len, align);
```

Comment:
We are surprised that enums are being treated as unsigned here, [because we thought they were supposed to be `int` for underlying type](https://en.cppreference.com/w/c/language/enum#Explanation) until C23.

We are hesitant to take the %d -> %u changes on enums blindly and instead would suggest finding a compiler flag that enforces what the standard says should happen here so we get consistency across compilers (this is otherwise likely something where compiler A says `unsigned` and B says `int` and it becomes a mess).

Reply:
Valid concern

The thing is enum types can be int type but they are not int in C. On gcc 1), enum types are unsigned int by default.
enum constants are int but enum types are implementation defined.
In your case the enum constant is int but you are giving it a value that does not fit in a int. You cannot have unsigned int enum constants in C as C says they are int.

I had to do some reading, from what I gathered, the type for `enum` is not that well defined, in C99, `enum` *constants* are int (C99 §6.4.4.3), but `enum` *types* are implementation defined (C99 §6.7.2.2/4), GCC defaults to `unsigned int` unless there are signed entries in the `enum` (https://gcc.gnu.org/onlinedocs/gcc/Structures-unions-enumerations-and-bit-fields-implementation.html), I didn't find any useful flags in this regard, but I'm looking at possible workarounds

Reply:
You raise a valid concern...

I had to do some reading, from what I gathered, the type for `enum` is not that well defined, in C99, `enum` *constants* are int (C99 §6.4.4.3), but `enum` *types* are implementation defined (C99 §6.7.2.2/4), GCC defaults to `unsigned int` unless there are signed entries in the `enum` (https://gcc.gnu.org/onlinedocs/gcc/Structures-unions-enumerations-and-bit-fields-implementation.html), I didn't find any useful flags in this regard, but I'm looking at possible workarounds

Reply:
No worries. Ok, given that we should probably check and see what both MSVC and Clang say on the subject (the former in particular because of Clang-CL which does what MSVC does rather than what Clang normally does) to try and pick out a path that will work into the future and not have us stumble on this problem again.

Then document what we choose and why - making an entry in HACKING.md for example - so we have a reference to go back to and a reasoning for other contributors to read for why we enforce whatever we end up enforcing.

Reply:
Enum changes were reverted for now, added a note to HACKING.md about format specifiers, this can be addressed in a future PR